index.d.ts

```diff
@@ -120,11 +120,17 @@ declare module "hyperapp" {
     | Action<S, P>
     | readonly [action: Action<S, P>, payload: P]
 
+  // An Effecter is the function that runs an effect
+  type Effecter<S, P = any> = (
+    dispatch: Dispatch<S>,
+    payload?: P
+  ) => void | Promise<void>
+
   // An effect is where side effects and any additional dispatching may occur.
-  type Effect<S, P = any> = readonly [
-    effecter: (dispatch: Dispatch<S>, payload: P) => void | Promise<void>,
-    payload: P
-  ]
+  type Effect<S, P = any> =
+    | Effecter<S, P>
+    | readonly [effecter: Effecter<S, P>, payload?: P]
```

Comment:
`payload?: P` doesn't pass my type tests but changing it back to `payload: P` will.